internal/controllers/suite_test.go

```diff
@@ -162,6 +162,15 @@ func TestMain(m *testing.M) {
 		ErrorIfCRDPathMissing: true,
 	}
 
+	// The BinaryAssetsDirectory is only necessary when we run tests directly
+	// without using the target. (e.g., when debugging tests in an IDE)
+	// If left unset, it will default to the path defined in controller-runtime (/usr/local/kubebuilder/).
+	// Ensure that required binaries are set up in the bin directory bin/k8s/envtest-binaries for direct testing.
+	// Run `make setup-envtest` to set up the binaries.
+	if getFirstFoundEnvTestBinaryDir() != "" {
```

Comment:
would it be cleaner if we just used an environment variable like ENV_TEST_BINARY_DIR or something like that?
Then, it doesn't really matter where the binaries are placed.

Reply:
By default, the binaries required for the project should be placed in the `bin` directory of the project.  

If we don’t enforce a specific location, then anyone wanting to debug tests **must**:  
- Understand that the tests require certain binaries.  
- Identify the correct environment variables needed.  
- Manually configure these variables every single time they want to debug something.  

By standardizing this, everything works seamlessly without requiring contributors to deeply understand or reconfigure the setup.  
I don’t need to set up test execution repeatedly—things “just work” as they should, hassle-free.

Reply:
We also need to take into account that bingo will eventually become obsolete and we will likely switch to `go tool` for managing tool dependencies once it lands in Go. See #341 and relevant Go issues mentioned there.

Reply:
This PR never was about BINGO
Sorry, I got lost in the journey and reviews and reverted it to solve what it should address and was addressing since the beginner.